android/src/main/kotlin/dev/steenbakker/mobile_scanner/MobileScanner.kt

```diff
@@ -462,6 +478,54 @@ class MobileScanner(
         }
     }
 
+    /**
+     * Inverts the image colours respecting the alpha channel
+     */
+    @SuppressLint("UnsafeOptInUsageError")
+    fun invertInputImage(imageProxy: ImageProxy): InputImage {
+        // Extract Image from ImageProxy
+        val image = imageProxy.image ?: throw IllegalArgumentException("Image is null")
+
+        // Convert YUV_420_888 image to NV21 format
+        val imageByteArray = yuv420888toNV21(image)
+
+        // Invert the cropped image
+        val invertedBytes = inverse(imageByteArray)
+
+        // Create a new InputImage from the inverted byte array
+        return InputImage.fromByteArray(
+            invertedBytes,
+            image.width,
+            image.height,
+            imageProxy.imageInfo.rotationDegrees,
+            InputImage.IMAGE_FORMAT_NV21
+        )
+    }
+
+    // Helper function to convert YUV_420_888 to NV21
+    private fun yuv420888toNV21(image: Image): ByteArray {
+        val yBuffer = image.planes[0].buffer
+        val uBuffer = image.planes[1].buffer
+        val vBuffer = image.planes[2].buffer
+
+        val ySize = yBuffer.remaining()
+        val uSize = uBuffer.remaining()
+        val vSize = vBuffer.remaining()
+
+        val nv21 = ByteArray(ySize + uSize + vSize)
+
+        yBuffer.get(nv21, 0, ySize)
+        vBuffer.get(nv21, ySize, vSize)
+        uBuffer.get(nv21, ySize + vSize, uSize)
+
+        return nv21
+    }
+
+    // Helper function to invert image data
```

Comment:
```suggestion
    private fun invertBytes(bytes: ByteArray): ByteArray {}
```

We can omit the comment if the method says what it does :)